### PR TITLE
Refactor ERC55 address support

### PR DESCRIPTION
**Pull request by Bushstar**

### CODE REVIEW

**Review by prasannavl — DISMISSED** [review]

**Comment by prasannavl** [comment]
Tests still need fixing. LGTM once sorted.

File: src/compressor.cpp
Comment (prasannavl) [inline src/compressor.cpp:25]: Can we instead do `CKeyID.GetRaw` that points to the raw hash and just use the address of that? 

Since memcpy shouldn't be relying on addresses returned by iterators. Iterator address can be different from the raw hashes.

Comment (prasannavl) [reply src/compressor.cpp:25]: Sure, I hear that - however `begin`, `end` is usually associated with iter interfaces and hence the nitpick to avoid a misrepresented convention.

File: src/outputtype.cpp
Comment (prasannavl) [inline src/outputtype.cpp:94]: Can't this be:

```
if (key.IsCompressed()) {
      pubkeyCopy.Decompress();
}
```

And just not un-equivocally use uncompressed for legacy and compressed for eth and bech32 without the branching?

File: src/script/signingprovider.cpp
Comment (prasannavl) [inline src/script/signingprovider.cpp:129]: Any specific reason to not just unconditionally store all 3 for easier maintainability?

Comment (Bushstar) [reply src/script/signingprovider.cpp:129]: We do store all three. Check the preceding line...

`mapKeys[pubkey.GetID()] = key;`

Comment (Bushstar) [reply src/script/signingprovider.cpp:129]: We then store whatever two was not already added below.

File: src/outputtype.cpp
Comment (Bushstar) [reply src/outputtype.cpp:94]: Sure. Will update and use everywhere.

**Review by prasannavl — DISMISSED** [review]

**Comment by prasannavl** [comment]
Have to resolve test failures, still before merging. Shaping up great!

**Review by prasannavl — DISMISSED** [review]

**Review by prasannavl — APPROVED** [review]

**Pull request merged**
